querydsl-codegen/src/main/java/com/querydsl/codegen/EntityType.java

```diff
@@ -50,30 +57,51 @@
 
     private final Map<Object, Object> data = new HashMap<Object,Object>();
 
-    private String uncapSimpleName;
+    private String modifiedSimpleName;
 
     /**
      * Create a new {@code EntityType} instance for the given type
      *
-     * @param type
+     * @param type the type to be used
      */
     public EntityType(Type type) {
+        this(type, new DefaultVariableNameFunction());
+    }
+
+    /**
+     * Create a new {@code EntityType} instance for the given type
+     *
+     * @param type the type to be used
+     * @param variableNameFunction the variable name function to be used
+     */
+    public EntityType(Type type, Function<EntityType, String> variableNameFunction) {
         this(type, new LinkedHashSet<Supertype>());
+        this.modifiedSimpleName = variableNameFunction.apply(this);
     }
 
     /**
      * Create a new {@code EntityType} instance for the given type and superTypes
      *
-     * @param type
-     * @param superTypes
+     * @param type the type to be used
+     * @param superTypes the super types to be used
      */
     public EntityType(Type type, Set<Supertype> superTypes) {
         super(type);
-        this.uncapSimpleName = StringUtils.uncapitalize(type.getSimpleName());
-        if (JavaSyntaxUtils.isReserved(uncapSimpleName)) {
-            this.uncapSimpleName = uncapSimpleName + "$";
-        }
         this.superTypes = superTypes;
+        this.modifiedSimpleName = new DefaultVariableNameFunction().apply(this);
```

Comment:
I've removed the unused constructor with three parameters.
We cannot remove the `new DefaultVariableNameFunction().apply(this)` from the `EntityType` constructor, unless we end up with merging the two constructors and changing all the calls in the code base. I would't see any benefit in that.